pkg/vectorstore/vectorstores.go

```diff
@@ -9,5 +9,5 @@ type VectorStore interface {
 	AddDocuments(ctx context.Context, docs []Document, collection string) ([]string, error)                      // @return documentIDs, error
 	SimilaritySearch(ctx context.Context, query string, numDocuments int, collection string) ([]Document, error) //nolint:lll
 	RemoveCollection(ctx context.Context, collection string) error
-	RemoveDocument(ctx context.Context, documentID string, collection string) error
+	RemoveDocument(ctx context.Context, documentID string, collection string, where, whereDocument map[string]string) error
```

Comment:
I left out the metadata filters because I wasn't sure if potential other VectorDBs would support them the same way. This is not really important to us right now, so I don't care too much.
On the other hand, we have the file metadata information and the file to documents mapping in the sqlite DB as well and thus could look up all document IDs there and then remove them directly by ID.
Both are valid ways.